client/landing/stepper/README.md

```diff
@@ -1,106 +1,402 @@
 	},
 };
 ```
 
-## The API
+#### Registering the flow
+
+Flows have to be registered [here](/client/landing/stepper/declarative-flow/registered-flows.ts).
+
+#### File hierarchy convention
 
-To create a flow, you only have to implement `initialize` and `useStepNavigation`. `initialize` can do any checks you need and it should finally return an array of step objects, `useStepNavigation` is the engine where you make navigation decisions. This hook returns an object of type [`NavigationControls`](./declarative-flow/internals/types.ts):
+[Please put your flow in a folder](https://dotcom.wordpress.com/2025/03/14/stepper-move-flows-into-their-own-folder/), not as a standalone file. And include a README file with cursory information about your flow **and testing steps**. Then simply link to them in your PRs (win-win).
 
-There is also an optional `useSideEffect` hook. You can implement this hook to run any side-effects to the flow. You can prefetch information, send track events when something changes, etc...
+#### Managing authentication
 
-There is a required `isSignupFlow` flag that _MUST be `true` for signup flows_ (generally where a new site may be created), and should be `false` for other flows. The `isSignupFlow` flag controls whether we'll trigger a `calypso_signup_start` Tracks event when the flow starts. For signup flows, you can also supply additional event props to the `calypso_signup_start` event by implementing the optional `useTracksEventProps()` hook on the flow.
+Stepper takes care of authenticating your users. You should not have to worry about auth at all. All you need to do is mark the steps as gated behind auth.
+
+```ts
+function initialize() {
+	// Gate all the steps
+	return stepsWithRequiredLogin( [ STEPS.UNIFIED_DOMAINS, STEPS.UNIFIED_PLANS, STEPS.PROCESSING ] );
+
+	// Gate some
+	return [ STEPS.UNIFIED_DOMAINS, ...stepsWithRequiredLogin( [ STEPS.UNIFIED_PLANS, STEPS.PROCESSING ] ) ] as const;
+}
+```
+
+**Stepper will take care of:**
+
+1. Injecting the user step when the user is not logged in.
+2. Sign up or log in the user.
+3. Bring the user back to the right step.
+
+#### Asserting conditions before running the flow
+
+If you want your flow to only be accessible to a certain type of user, you can assert the conditions in the `initialize` function.
+
+```ts
+async function initialize() {
+	const hasAccessToFlow = await isAgencyUser();
+
+	if ( ! hasAccessToFlow ) {
+		window.location = '/home';
+		// Return false to halt the fetching and execution of the steps. This makes the redirect much faster.
+		return false;
+	}
+
+	return [ STEPS.UNIFIED_PLANS, STEPS.PROCESSING ] as const;
+}
+```
+
+#### Pending actions
+
+### Notes
+
+1. A successful flow is a rare event, so most flows are sadly ephemeral. Please keep as much logic, CSS, and code as possible into the flow folder itself. So that when it's deleted, clean up is easy.
+2. Adding testing steps in your flow's README goes a long way. It would be really appreciated.
```

Comment:
I would be a bit more prescriptive about adding:

- docs
- clear ownership
- tests